tests/Integration/WordpressTest.php

```diff
@@ -21,4 +21,51 @@ public function test_it_can_sen_error_notifications_to_the_admin()
         $this->assertSame("subject", $mail["subject"]);
         $this->assertSame("admin@test.com", $mail["to"][0][0]);
     }
+
+    public function test_it_can_add_a_cron_interval()
+    {
+        WordPress::addCronInterval("key", 5, "Display Name");
+
+        $schedules = \wp_get_schedules();
+        $this->assertNotNull($schedules["key"]);
+        $schedule = $schedules["key"];
+        $this->assertSame(5, $schedule["interval"]);
+        $this->assertSame("Display Name", $schedule["display"]);
+    }
+
+    public function test_it_can_schedule_a_cron_job()
+    {
+        WordPress::scheduleCronjob("hook", strtotime("now"), "hourly");
+
+        $event = \wp_get_schedule("hook");
+        $this->assertSame("hourly", $event);
+    }
+
+    public function test_it_does_not_schedule_cron_events_multiple_times()
+    {
+        WordPress::scheduleCronjob("hook", strtotime("now"), "hourly");
+        WordPress::scheduleCronjob("hook", time() + 3600, "daily");
+
+        $event = \wp_get_schedule("hook");
+        $this->assertSame("hourly", $event);
+    }
+
+    public function test_it_can_add_an_action_for_a_cronjob()
+    {
+        global $wp_filter;
+
+        WordPress::addCronjobAction("hook", function () {
+        });
+
+        $this->assertCount(1, $wp_filter["hook"][10]);
```

Comment:
just curious why it's exactly item #10? could this order be different on some other wp version or even a different system ?

Reply:
Hooks can have different priorities. At positition 10 in the array for the hook, there is an array containing all hook with priority 10. Since 10 is the default priority (and we do not allow to specify any other priority) we have to check if the function for the hook has been added to this array.

Reply:
thanks